src/Index.js

```diff
@@ -652,12 +652,109 @@ Index.prototype.getSettings = function(callback) {
   var indexObj = this;
   return this.as._jsonRequest({
     method: 'GET',
-    url: '/1/indexes/' + encodeURIComponent(indexObj.indexName) + '/settings',
+    url: '/1/indexes/' + encodeURIComponent(indexObj.indexName) + '/settings?getVersion=2',
```

Comment:
> Any client that uses getSettings to get synonyms in some sync scripts

Can you explain the issue a bit more? Every method call is already asynchronous by default, then if getVersion=2 triggers a different behavior than before, I was not aware of that.